File: 10-stacks-queues1/2-stack-operations/main.java
// Stack Operations - Java Example

import java.util.Stack;

public class StackOperations {
    public static void main(String[] args) {
        Stack<Integer> stack = new Stack<>();

        stack.push(1);
        stack.push(2);

        if (!stack.isEmpty())
            System.out.println("Top element: " + stack.peek());

        if (!stack.isEmpty())
            System.out.println("Pop element: " + stack.pop());

        System.out.println("Is stack empty? " + stack.isEmpty());
    }
}

File: 10-stacks-queues1/2-stack-operations/main.py
# Stack Operations - Python Example

stack = []

def push(val):
    stack.append(val)

def pop():
    if stack:
        return stack.pop()
    return None

def peek():
    if stack:
        return stack[-1]
    return None

def is_empty():
    return len(stack) == 0

push(1)
push(2)
print("Top element:", peek())
print("Pop element:", pop())
print("Is stack empty?", is_empty())
